src/main/java/cd/go/contrib/elasticagent/ElasticProfileSettings.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cd.go.contrib.elasticagent;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.Period;
+
+public class ElasticProfileSettings {
+    private String goServerUrl;
```

Comment:
I agree with you. 
However I got ahead of myself. I had another change planned . I will add the front end code to capture these and after that it will be very easy to have multi cluster capability.There should not be any more changes required in the backend since the I am creating multiple client for each namespace anyways.

Reply:
Added the front end code to capture Cluster Url and CA Cert data now

Reply:
So, with this approach, there will be a cluster defined at the plugin-settings level. But, one can override the cluster information at the elastic-profile level? Is my understanding correct?

Reply:
Yes. That is correct. With this the profiles can support multiple clusters

Reply:
@saifally -- It will work, but it leads to a lot of duplication of settings on all the profiles.
Also, this makes the plugin-settings not very useful as all the properties of plugin-settings could be overridden/defined at the profile level.

How about:
* Allowing users to define named-clusters in plugin settings. The data-structure of cluster changes to the collection of named clusters.
* These named-clusters can be associated with the elastic profiles.
* As users can define as many clusters, I don't see a need of allowing users to override the namespace and Security Token.
* Auto-register-timeout is not a cluster-property and depends on the agent-image used. Hence auto-register-timeout should be allowed to override at the profile level.

We can,
* validate the elastic profile to see if the referenced cluster exists.
* Show read-only information about the referenced cluster at the profile.

\cc: @bdpiparva

Reply:
First of all , I am not against your idea. I see the value in it.

Few thoughts though :)

-  For my approach , plugin level settings is the default settings. For people who want to use One Cluster and One Namespace, that is the right place. So it is still relevant and useful
- If we add multiple clusters at the plugin level. It will start looking cluttered after a while,unless the UI is designed right  I am not a front end/UX expert so I wont be able to do justice to this.
Happy to contribute backend code to a branch if you guys can pitch in with front end stuff.
- For One Cluster - Multiple Namespaces , this approach of override what you need to ,at Profile level has good user experience (in my opinion.)  We are not forcing the user to setup something on one screen and then go to some other place and use that.

However keen to hear thoughts from you and your team.
After-all you guys have to be consistent across Kubernetes, Marathon,Swarm and ECS Elastic Agent plugin

Reply:
One quick thing about the duplication of settings . For my approach, if Namespace,Token,Cluster URL and Cluster CA are not specified at Elastic Profile Level.,they are not saved at that level on save of Elastic Profile.

They are merged with Plugin settings only when a KubernetesInstance is created,i.e. an elastic agent is created.

Reply:
@saifally -- Apology for the delayed response.
We can go ahead with the approach of specifying cluster details at plugin-settings level and if anyone wants, they can override it at the elastic-profile level.

Implementation specific:
We can define all the cluster attributes at the elastic-profile level (view, model, etc). And prior to creating an agent, we can merge the settings with the elastic profile.
For the `create-agent` call, we should still pass in `elastic-profile` (currently, we are passing settings).

Also, we need to redesign the status report page, considering multi-cluster.

Please let us know if we can help you with any of the implementation stuff. :)

Reply:
@GaneshSPatil  Thanks a lot for agreeing to my suggestions.

For the implementation details can you add the comment on the piece of code.
I am not very clear on the changes you are mentioning.

Good point about the status page.
For now I have added namespace to the status report page.
Can I add that as part of another pull request or are you keen for that change in the same pull request

Reply:
> For the implementation details can you add the comment on the piece of code.

Sure. will do.

> Can I add that as part of another pull request or are you keen for that change in the same pull request

Sure, we can get the main functionality implemented as part of this PR. and subsequent PRs can address other relevant changes.

Reply:
@GaneshSPatil I have refactored the code. To use elastic profile instead of settings.
Have a look when you get time.